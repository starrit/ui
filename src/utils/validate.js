import React from 'react';
import constants from 'constants/app-constants';

export const required = value => {
  if (!value.toString().trim().length) {
    // We can return string or jsx as the 'error' prop for the validated Component
    return <div className="field-error"><p>This field is required</p></div>;
  }
};

const isValidEmail = emailString => constants.REGEX.EMAIL.test(emailString);
export const email = value => {
  if (!isValidEmail(value)) {
    // We can return string or jsx as the 'error' prop for the validated Component
    return <div className="field-error"><p>Provide a valid email</p></div>;
  }
};

export const passwordConstraints = value => {
  if (value && !constants.REGEX.PASSWORD.test(value)) {
    return <div className="field-error"><p>
      Password should be atleast 8 characters long and contain atleast 1 number
    </p></div>;
  }
};

export const confirmPassword = (value, props, components) => {
  let passwordValue = components.password[0].value;
  let message;
  if (value && !passwordValue) {
    message = 'Confirm password needs to by empty for empty password';
  }
  else if (!value && passwordValue) {
    message = 'This field is required';
  }
  else if (value && passwordValue && value !== passwordValue) {
    message = 'Passwords need to match';
  }
  if (message) {
    return <div className="field-error"><p>{message}</p></div>;
  }
};


export default {
  isValidEmail
};
