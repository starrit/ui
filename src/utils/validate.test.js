import validate, {required} from './validate';
import renderer from 'react-test-renderer';

describe('Validate Util Functions', () => {
  it('should validate email pattern', () => {
    expect(validate.isValidEmail('test@example.com')).toBeTruthy();
  });

  it('should return jsx element with error message', () => {
    const tree = renderer.create(required('')).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should not return anything', () => {
    const tree = renderer.create(required('Some string here')).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
