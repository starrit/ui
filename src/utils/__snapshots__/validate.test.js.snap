// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validate Util Functions should not return anything 1`] = `null`;

exports[`Validate Util Functions should return jsx element with error message 1`] = `
<div
  className="field-error"
>
  <p>
    This field is required
  </p>
</div>
`;
